apollo-router/src/introspection.rs

```diff
@@ -138,32 +159,47 @@ impl IntrospectionCache {
         let schema = schema.clone();
         let doc = doc.clone();
         let priority = compute_job::Priority::P1; // Low priority
-        let response =
-            compute_job::execute(priority, move || Self::execute_introspection(&schema, &doc))
-                .await
-                .expect("Introspection panicked");
+        let response = compute_job::execute(priority, move || {
+            Self::execute_introspection(max_depth, &schema, &doc)
+        })
+        .await
+        .expect("Introspection panicked");
```

Comment:
Do we have to keep this `.expect` here, or can it be an error?

Reply:
I believe the error path here is unreachable. `compute_job::execute` would return `Err` if a job panics, but before it gets a chance to do so the panic handler in `apollo-router/src/executable.rs` would exit the process. This leaves `expect` only for type-checking.

But even if we didn’t have this handler, what makes this path special out of all the places we could potentially catch panics and turn them into errors?

Reply:
It would be helpful to add that context about the panic handler as a code comment for future readers